lib/url_to_subscriber_list_criteria.rb

```diff
@@ -0,0 +1,157 @@
+require 'uri'
+
+class UrlToSubscriberListCriteria
+  MISSING_LOOKUP = "*** MISSING KEY ***"
+  EMAIL_SUPERTYPE = "email_document_supertype"
+  GOVERNMENT_SUPERTYPE = "government_document_supertype"
+  class UnprocessableUrl < StandardError; end
+
+  def initialize(url, static_data = StaticData)
+    @url = URI.parse(url.strip)
+    @static_data = static_data
+    @missing_lookups = []
+  end
+
+  def convert
+    @convert ||= begin
+      hash = map_url_to_hash.dup
+      if hash["links"]
+        links = hash["links"].each_with_object({}) do |(key, values), result|
+          result[key] = values.map { |value| lookup_content_id(key, value) }
+        end
+        hash.merge!("links" => links)
+      end
+      hash
+    end
+  end
+
+  def missing_lookup
+    convert
+    @missing_lookups.to_sentence if @missing_lookups.any?
+  end
+
+  def map_url_to_hash
+    @map_url_to_hash ||= begin
+      result = if @url.path.match(%r{^/government/statistics\.atom$})
+                 { "links" => from_params, EMAIL_SUPERTYPE => "publications", GOVERNMENT_SUPERTYPE => "statistics" }
+               elsif @url.path.match(%r{^/government/publications\.atom$})
+                 { "links" => from_params, EMAIL_SUPERTYPE => "publications" }
+               elsif @url.path.match(%r{^/government/announcements\.atom$})
+                 { "links" => from_params, EMAIL_SUPERTYPE => "announcements" }
+               elsif (path_match = @url.path.match(%r{^/government/world/(.*)\.atom$}))
```

Comment:
I remembered that PageWatch is enabled on these world location topics, and wondered whether we should migrate them across - I thought that maybe Whitehall isn’t sending those emails and so we shouldn’t start sending them from email-alert-api. It looks like Whitehall is sending notifications to these lists, though - which means that lots of them are duplicated a few minutes later when PageWatch picks it up :( We should keep the world location lists in the migration - we aren’t going to make the duplication worse by doing this.

We should look at disabling PageWatch on the world location feeds - so far I’ve found very few things that are sent by PageWatch that we don’t also send, and migrating to email-alert-api may well fill those gaps anyway. It could do with more investigation before we make that decision, though.